cmake/compiler/xcc/compiler_flags.cmake

```diff
@@ -1,6 +1,10 @@
 # No special flags are needed for xcc.
 # Only select whether gcc or clang flags should be inherited.
 if(CC STREQUAL "clang")
+  if($ENV{XCC_NO_G_FLAG})
```

Comment:
why does we need an undocumented environment setting for this ?
adding to this mess: 
https://github.com/zephyrproject-rtos/zephyr/issues/40389

wouldn't it be better to add a Kconfig setting that allows users to disable debug information, and then safeguard the CMake code here with that Kconfig setting:
https://github.com/zephyrproject-rtos/zephyr/blob/b374dc6a7fd8f841394ad6cf2364c7ceef05f3c7/CMakeLists.txt#L265

Reply:
Usage of "-g" only results in errors with older XCC version (earlier than 2018) with Clang 3.9.0. Newer XCC version, e.g. RI-2018.0, no longer has this issue even though its Clang is still 3.9.0. The GCC part of XCC does not have this issue. There is no reliable way to determine whether XCC/GCC or XCC/Clang is being used under kconfig so the new kconfig cannot be created with a XCC/Clang specific default, not to mention trying to determine XCC version there. So it seems using environment variables here is the least invasive compromise to achieve what needs to be done.

Reply:
> There is no reliable way to determine whether XCC/GCC or XCC/Clang is being used under kconfig so the new kconfig cannot be created with a XCC/Clang specific default,

Correct, but the same goes for `XCC_NO_G_FLAG`, there is no `XCC/Clang` default for the environment flag.
User must manually set this flag, just as user would have to disable debug information, like `-DCONFIG_DEBUG_INFO=n`, if we went the Kconfig direction.

Benefit of Kconfig is that it is easier for a user to lookup and it's always documented when `help` field is written.

The only benefit I see with regards to the env variable is the fact that for those switching between toolchains, then it can be set once, and only impact xcc, and not other toolchains.

This is just like the: [XCC_USE_CLANG](https://github.com/zephyrproject-rtos/zephyr/blob/20c1e1a8000b3da7817f670825178391a9649016/cmake/toolchain/xcc/generic.cmake#L26) setting that is also not documented anywhere, which I also mention in this issue: #40389

> So it seems using environment variables here is the least invasive compromise to achieve what needs to be done.

Partly agree, for those who knows about this setting it appears less intrusive, but it is very hidden for others who might face this issue, that such an environment variable exists.

And imagine if every toolchain started to just randomly add new env settings without documenting those, then it would be an even worse mess.

Reply:
btw, when looking at the only documentation we have regarding `Xtensa` / `xcc` toolchain here:
https://docs.zephyrproject.org/latest/boards/xtensa/intel_s1000_crb/doc/index.html#set-up-build-environment

Then I see users are required to specify: 
```
export TOOLCHAIN_VER=RI-2018.0-linux
```

isn't that piece of information together with `clang` knowledge enough to determine whether or not the toolchain supports `-g` for assembly files ?

Reply:
The thing is that I have no way to confirm when the fix was incorporated. "-g" is no longer an issue in RI-2018.0 for s1000 by simply building with it. There may be a 2018.1 version which is built without the fix. The environment variable is simply there to let developers decide what needs to be done instead of forcing a default which may not be correct (and then will need something else to override it).

Anyway, I think we definitely need some pages in the documentation for non-SDK toolchains.

Reply:
> The thing is that I have no way to confirm when the fix was incorporated. "-g" is no longer an issue in RI-2018.0 for s1000 by simply building with it.

Fair enough. Just asking as I don't have a license for that toolchain, so I cannot verify behavior and / or what info is available.

Sounds like long term we should get asm compiler testing implemented here:
https://github.com/zephyrproject-rtos/zephyr/blob/becc1efe512212bd85b9ec25a6e97ed61952a633/cmake/extensions.cmake#L1775-L1781

I believe that should solve this particular issue.

> Anyway, I think we definitely need some pages in the documentation for non-SDK toolchains.

For 3rd party toolchains we have this page: https://docs.zephyrproject.org/latest/getting_started/toolchain_3rd_party_x_compilers.html

but it appears no one never bothered to move the Xtensa / xcc toolchain from here https://docs.zephyrproject.org/latest/boards/xtensa/intel_s1000_crb/doc/index.html#set-up-build-environment to that location instead.

I think the Xtensa toolchain description would benefit a lot if it is relocated, and then updated with the two hidden environment variables now available.

Reply:
Adding a few words here: https://github.com/zephyrproject-rtos/zephyr/pull/42378

The s1000 is specified to the board with some additional bits on debug tool setups. I think it would be better to leave it there so the all relevant information is contained in one page.

Reply:
great, thanks.